### PR TITLE
MAINT-28042 : Fix problem of loading add News button

### PR DESCRIPTION
**Pull request by ahamdi**
This PR fixes two problems : 
 - Make sure to add the extension asynchronously to composer, and load resource bundle when  refreshing extensions list
 - Use a function  SpaceUtils.isRedactor to check if user is redactor in that space

### CODE REVIEW

File: component/core/src/main/java/org/exoplatform/social/core/space/SpaceUtils.java [hunk 1]
```diff
@@ -1791,6 +1792,32 @@ public static boolean isUserHasMembershipTypesInGroup(String remoteId, String gr
...
+   * @throws Exception
+   */
+  public static boolean isRedactor (String userName, String spaceGroupId) throws Exception{
+    boolean spaceHasRedactorRole = false;
+    String REDACTOR_MEMBERSHIP_NAME = "redactor";
+
+    List<MembershipType> membershipTypes = getOrganizationService().getMembershipHandler().findMembershipTypesByGroup(spaceGroupId);
+    for(MembershipType membershipType : membershipTypes) {
+      if(REDACTOR_MEMBERSHIP_NAME.equals(membershipType.getName())) {
+        spaceHasRedactorRole = true;
```
Comment (azayati) [inline component/core/src/main/java/org/exoplatform/social/core/space/SpaceUtils.java]: reeturn getOrganizationService().getMembershipHandler().
              findMembershipByUserGroupAndType(userName, spaceGroupId, REDACTOR_MEMBERSHIP_NAME) != null;

File: component/core/src/main/java/org/exoplatform/social/core/space/SpaceUtils.java [hunk 2]
```diff
@@ -1791,6 +1792,32 @@ public static boolean isUserHasMembershipTypesInGroup(String remoteId, String gr
...
+        break;
+      }
+    }
```
Comment (azayati) [inline component/core/src/main/java/org/exoplatform/social/core/space/SpaceUtils.java:1812]: return true

File: component/core/src/main/java/org/exoplatform/social/core/space/SpaceUtils.java [hunk 3]
```diff
@@ -1791,6 +1792,32 @@ public static boolean isUserHasMembershipTypesInGroup(String remoteId, String gr
...
+    if (spaceHasRedactorRole) {
```
Comment (azayati) [inline component/core/src/main/java/org/exoplatform/social/core/space/SpaceUtils.java]: Not needed IMO

File: extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl [hunk 4]
```diff
@@ -31,20 +31,14 @@
   ExoFeatureService featureService = ExoContainerContext.getCurrentContainer().getComponentInstanceOfType(ExoFeatureService.class);
   boolean isMainStreamComposerEnabled = System.getProperty("exo.config.mainStreamComposer.enabled").equals("true");
   //check if space has redactors
-  Boolean spaceHasRedactors = false;
   Boolean isCurrentUserRedactor = false;
```
Comment (azayati) [inline extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl:34]: Boolean isCurrentUserRedactor = isMainStreamComposerEnabled ;

File: extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl [hunk 5]
```diff
@@ -31,20 +31,14 @@
...
-      //check if connected user is redactor
-      String currentUserId = _ctx.getRequestContext().getRemoteUser();
-      isCurrentUserRedactor = spaceRedactors.contains(currentUserId);
-    }
+    String currentUserId = _ctx.getRequestContext().getRemoteUser();
+    isCurrentUserRedactor = SpaceUtils.isRedactor(currentUserId, spaceGroupId);
   }
   if(featureService.isActiveFeature("new-composer")) {
-    if (uicomponent.getSpaceGroupId() != null && (!spaceHasRedactors || isCurrentUserRedactor) || (isMainStreamComposerEnabled && uicomponent.getSpaceGroupId() == null)) {
+    if (uicomponent.getSpaceGroupId() != null && isCurrentUserRedactor || (isMainStreamComposerEnabled && uicomponent.getSpaceGroupId() == null)) {
```
Comment (azayati) [inline extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl:41]:
```suggestion
    if (isCurrentUserRedactor) {
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl (hunk 5 above)
Comment (ahamdi) [reply extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl:41]: I don't get why you do want to change this logic as it is out of the scope of this PR.
This is related to the case where we are on the Main activity stream (outside a space) and should not be changed in the scope of this PR

File: component/core/src/main/java/org/exoplatform/social/core/space/SpaceUtils.java (hunk 2 above)
Comment (ahamdi) [reply component/core/src/main/java/org/exoplatform/social/core/space/SpaceUtils.java:1812]: done

File: extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl (hunk 4 above)
Comment (ahamdi) [reply extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl:34]: No they are different

**Review by azayati — APPROVED** [review]
LGTM after fixing conflicts

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**